tests/sqlparser_common.rs

```diff
@@ -2259,18 +2263,34 @@ fn parse_agg_with_order_by() {
             Box::new(MsSqlDialect {}),
             Box::new(AnsiDialect {}),
             Box::new(HiveDialect {}),
+            Box::new(SnowflakeDialect {}),
         ],
         options: None,
     };
 
     for sql in [
```

Comment:
Why were these tests changed?

This PR adds support for window functions (aka those that have an `OVER` clause) which is different than the `ORDER BY` clause in the aggregate function argument.

I think you should keep the existing tests and add a new one for the IGNORE NULLS / RESPECT NULLs syntax

Reply:
Thanks for the feedback @alamb .
Please look at the syntax of `FIRST_VALUE` (for example) in some of the dialects:
https://docs.snowflake.com/en/sql-reference/functions/first_value
https://learn.microsoft.com/en-us/sql/t-sql/functions/first-value-transact-sql?view=sql-server-ver16
https://www.postgresqltutorial.com/postgresql-window-function/postgresql-first_value-function/

The `Over` Keyword is a must. so the syntax `FIRST_VALUE(x ORDER BY x)` is not valid.
Even if the tests pass, those tests are not aligned with the syntax.
That's why I added two arrays of dialects - for each dialect his supported syntax.

Reply:
> The Over Keyword is a must. so the syntax FIRST_VALUE(x ORDER BY x) is not valid.

I think that is true when the FIRST_VALUE function is being used as a window function

Some systems allow `FIRST_VALUE` to be used as normal aggregate functions -- so like

```sql
SELECT 
  FIRST_VALUE(amount ORDER BY time), 
  LAST_VALUE(amount ORDER BY time)
FROM 
  t
GROUP BY
  currency
```

Perhaps @mustafasrepo  can offer some perspective as the author of https://github.com/sqlparser-rs/sqlparser-rs/pull/882

But basically I think we should retain these tests

Reply:
Actually `FIRST_VALUE` and `LAST_VALUE` can be used as window or aggregate function (similar to `SUM`, `MIN`, etc.). In these tests, they are used as aggregate function. Hence these tests should remain as is.

Reply:
Thanks @alamb  and @mustafasrepo :) Learned something about these functions :)
I maintained the old tests